frontend/pages/hosts/HostDetailsPage/TransferHostModal/TransferHostModal.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useState } from "react";
+import { Link } from "react-router";
+import PATHS from "router/paths";
+import Modal from "components/modals/Modal";
+import Button from "components/buttons/Button";
+// ignore TS error for now until these are rewritten in ts.
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import { ITeam } from "interfaces/team";
+import { IHost } from "interfaces/host";
+
+interface ITransferHostModal {
+  isGlobalAdmin: boolean;
+  teams: ITeam[];
+  onSubmit: (team: ITeam) => void;
+  onCancel: () => void;
+  host: IHost;
+}
+
+interface INoTeamOption {
+  id: string;
+}
+
+const baseClass = "transfer-host-modal";
+
+const NO_TEAM_OPTION = {
+  value: "no-team",
+  label: "No team",
+};
+
+const TransferHostModal = (props: ITransferHostModal): JSX.Element => {
+  const { onCancel, onSubmit, teams, isGlobalAdmin, host } = props;
```

Comment:
Just a nugget of info: You can also destructure the props in the component's signature. So you could do
```
const TransferHostModal = ({ onCancel, onSubmit, teams, isGlobalAdmin, host }: ITransferHostModal): JSX.Element...
```

Reply:
That's pretty cool!